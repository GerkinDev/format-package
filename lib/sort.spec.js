const sort = require('./sort');

describe('sort', () => {
  let order;
  let keys;
  let restKey;

  beforeEach(() => {
    restKey = '...rest';
    keys = ['foo', 'bar', 'baz', 'qux'];
    order = ['foo', '...rest', 'baz'];
  });

  it('orders relevant keys', () => {
    expect(sort(keys, order)).toMatchSnapshot();
  });

  it('allows any key to be used to denote the "rest"', () => {
    restKey = '...extra';
    order = ['baz', 'foo', '...extra'];
    expect(sort(keys, order, restKey)).toMatchSnapshot();
  });

  it('appends extra keys to the end in alphabetical order if no "rest" is found', () => {
    order = ['qux'];
    expect(sort(keys, order)).toMatchSnapshot();
  });
});
