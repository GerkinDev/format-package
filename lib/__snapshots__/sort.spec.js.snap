// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sort allows any key to be used to denote the "rest" 1`] = `
Array [
  "baz",
  "foo",
  "bar",
  "qux",
]
`;

exports[`sort appends extra keys to the end in alphabetical order if no "rest" is found 1`] = `
Array [
  "qux",
  "bar",
  "baz",
  "foo",
]
`;

exports[`sort orders relevant keys 1`] = `
Array [
  "foo",
  "bar",
  "qux",
  "baz",
]
`;
